etc/nginx/sites-available/wazo

```diff
@@ -18,7 +18,7 @@ server {
     error_log /var/log/nginx/wazo.error.log;
     root /var/www/html;
 
-    include /etc/nginx/locations/http-enabled/*;
+    return 301 https://$host$request_uri;
```

Comment:
Unfortunately, this breaks phone directories over HTTP URLs, which are configured in `etc/nginx/locations/http-enabled/wazo-phoned`. Some phones do not support HTTPS, so we are stuck with HTTP on this one.

This also breaks the `http://wazo.example.com/api/` URL that is configured in `etc/nginx/locations/http-enabled/xivo-swagger-doc`. I don't know why this URL is redirected to HTTP-only...

My suggestion is to leave this configuration as-is, and change the files in `etc/nginx/locations/http-enabled` if possible.

Reply:
> Unfortunately, this breaks phone directories over HTTP URLs, which are configured in `etc/nginx/locations/http-enabled/wazo-phoned`. Some phones do not support HTTPS, so we are stuck with HTTP on this one.

Strange, I do not have `/etc/nginx/locations/http-enabled/wazo-phoned` here on my stacks.
Phone directories are managed by `wazo-phoned` on port 9498, so they should not be impacted by this port 80 modification ?

> This also breaks the `http://wazo.example.com/api/` URL that is configured in `etc/nginx/locations/http-enabled/xivo-swagger-doc`. I don't know why this URL is redirected to HTTP-only...

I tested it through HTTPS, and it works flawlessly.
Perhaps we could thing about migrating this to HTTPS ?
I'm not sure there's a GitHub repo to handle this modification, could you then perform it ?

Reply:
> > Unfortunately, this breaks phone directories over HTTP URLs, which are configured in `etc/nginx/locations/http-enabled/wazo-phoned`. Some phones do not support HTTPS, so we are stuck with HTTP on this one.
> 
> Strange, I do not have `/etc/nginx/locations/http-enabled/wazo-phoned` here on my stacks. Phone directories are managed by `wazo-phoned` on port 9498, so they should not be impacted by this port 80 modification ?

You're absolutely right, sorry about that, it was a test file that I had left on my own server...
Objection dismissed then :)

> > This also breaks the `http://wazo.example.com/api/` URL that is configured in `etc/nginx/locations/http-enabled/xivo-swagger-doc`. I don't know why this URL is redirected to HTTP-only...
> 
> I tested it through HTTPS, and it works flawlessly. Perhaps we could thing about migrating this to HTTPS ? I'm not sure there's a GitHub repo to handle this modification, could you then perform it ?

Yes, migrating to HTTPS seems like the right way. The repo for this file is here: https://github.com/wazo-platform/xivo-swagger-doc. I don't have the time to make the modification right now, sorry.

Reply:
Thank you again for your feedback.
Xivo Swagger Doc moved to HTTPS in https://github.com/wazo-platform/xivo-swagger-doc/pull/12 👍

Reply:
After digging some more, I think HSTS will make things harder in some setups: when HSTS is enabled, one cannot use `https://<ip_address>` anymore.

So HSTS can only be used once the server has a domain name, and since Wazo Platform does not enforce having a domain name, it can break some use cases where the server is only accessed via IP address.

As a consequence, I think we should not enforce HSTS by default, but maybe make it easy to enable... I don't know exactly what kind of mechanism could be used though...

Maybe we can extract the HSTS part to another PR, so that we can merge the rest?